chocolatey/NuGet.Client
# (#9) Check whether FindPackagesById is supported for a feed

## Description Of Changes

When some sources doesn't support the endpoint `FindPackagesById` we need to be able to fall back to a different endpoint to get the results that we are interested in.
The changes in this PR makes the necessary changes to check if a feed reports that it supports `FindPackagesById`, and if it does not explicitly report it is supported we will fall back to making use of the `Packages` endpoint instead by passing in the necessary filters that would be the equivalent to how FindPackagesById functions.

This has been implemented on all v2 related calls to a method that would result in a call to that endpoint (those that could be found).

## Motivation and Context

To be able to install dependent packages even from feeds that does not support `FindPackagesById`.

## Testing

1. Check out this PR, and build the NuGet.Client solution
2. Pull the changes into Chocolatey source code
3. Open fiddler on your machine
4. Ensure that you have a valid license file in place in debug version of Chocolatey
5. Run the following command `install chocolatey-agent --source=chocolatey.licensed --proxy=http://localhost:8888`
6. Verify that in the emitted query for the dependency `chocolatey.extension` that it does not make use of `FindPackagesById`.
7. Run the following command `install 7zip --proxy=http://localhost:8888 --source chocolatey`
8. Verify that in the emitted query for the depentdency `7zip.install` that it does make use of `FindPackagesById`

### Operating Systems Testing

- Windows 10/11

## Change Types Made


## Change Checklist


## Related Issues

- #9  
- https://app.clickup.com/t/20540031/PROJ-521
